src/vmm/src/lib.rs

```diff
@@ -600,6 +603,50 @@ impl Vmm {
         Ok(cpu_configs)
     }
 
+    /// Adds new vCPUs to VMM.
+    pub fn hotplug_vcpus(&mut self, config: HotplugVcpuConfig) -> Result<(), HotplugVcpuError> {
+        #[allow(clippy::cast_possible_truncation)]
+        if config.vcpu_count < 1 {
+            return Err(HotplugVcpuError::VcpuCountTooLow);
+        } else if self.vcpus_handles.len() as u8 + config.vcpu_count > MAX_SUPPORTED_VCPUS {
```

Comment:
This _could_ overflow and wrap around, e.g. if `self.vcpu_handles.len() == 2` and `config.vcpu_count == 255`! Let's add a test case for this and fix it :)